### PR TITLE
Add runner engine for JSON test files

### PR DESCRIPTION
**Pull request by VictorGrycuk**
## Proposed Changes
Add `runTest()s` function to load and run tests from JSON files.

## Testing
The test inside the file `with 'add' verb.json` must pass

### CODE REVIEW

File: packages/LUIS/test/The LUIS cli tool/with 'add' verb.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[
```
Comment (JuanAr) [inline packages/LUIS/test/The LUIS cli tool/with 'add' verb.json]: I'd suggest moving the test folder hierarchy to json. e.g.
```
{
  "suite" : {
    "title" : "The LUIS cli tool",
    "test"  {
        "prefix" : "with 'add' verb",
      [
        {
          "title" : "and 'application' command",
          "args" : ...
        }
      ]
  }
}
```

File: packages/LUIS/test/luis.cli.test.suite.js [hunk 2]
```diff
@@ -41,4 +42,6 @@ describe('The LUIS cli tool', () => {
         });
         
     });
+
+    runTests(__dirname);
```
Comment (JuanAr) [inline packages/LUIS/test/luis.cli.test.suite.js]: I'd suggest running tests within a single file. Meaning, not all json files but a relative single .json filename passed as parameter.

File: packages/LUIS/test/utils.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        describe(testGroup, () => {
+            GetTestGroup(directory);
+        });
+    });
+};
+
+function GetTestGroup(directory) {
+    const fs = require('fs');
+    readdirSync(directory).forEach(file => {
+        describe(file.slice(0, -5), () => {
```
Comment (JuanAr) [inline packages/LUIS/test/utils.js]: Come on! 👎

**Comment by JuanAr** [comment]
I demand you two to cross review your PRs from now on. Merciless review.

File: packages/LUIS/test/utils.js (hunk 3 above)
Comment (JuanAr) [reply packages/LUIS/test/utils.js]: I just stopped reviewing after I saw this.

File: packages/LUIS/test/utils.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        const testGroup = directory.split(path.sep).pop()
+        describe(testGroup, () => {
+            GetTestGroup(directory);
+        });
+    });
+};
+
+function GetTestGroup(directory) {
+    const fs = require('fs');
+    readdirSync(directory).forEach(file => {
```
Comment (JuanAr) [inline packages/LUIS/test/utils.js]: There's no .json filtering or something

File: packages/LUIS/test/The LUIS cli tool/with 'add' verb.json (hunk 1 above)
Comment (VictorGrycuk) [reply packages/LUIS/test/The LUIS cli tool/with 'add' verb.json]: I would not suggest grouping all the tests for a suite within a single json file.
For `luis.cli.test.suite.js` alone we are talking about over 126, counting only just a single branching for each command line.
The number of tests could easily triplicate o quadruplicate depending how thorough one wants the coverage.

**Comment by VictorGrycuk** [comment]
> The approach is good. Needs some refinement.
> Why having a mixture of coded and 'automated' tests?
> Convert all existing to automated.

@JuanAr It was agreed to leave intact the existing tests code, and only add new tests as json files. Hence the extraction to an utils file to modify the least amount of lines of the existing test files.

The validation for file extension and a more flexible approach to remove extension from file names was added.
However, these implementation will soon change due to the modifications to the json files.

**Comment by JuanAr** [comment]
Will continue review on top of https://github.com/southworkscom/botbuilder-tools/pull/81

**Pull request closed without merging**